twine/__main__.py

```diff
@@ -12,20 +12,37 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import argparse
 import sys
 from typing import Any
 
+import colorama
 import requests
 
 from twine import cli
 from twine import exceptions
+from twine import settings
 
 
 def main() -> Any:
     try:
         return cli.dispatch(sys.argv[1:])
     except (exceptions.TwineException, requests.HTTPError) as exc:
-        return "{}: {}".format(exc.__class__.__name__, exc.args[0])
+        return _format_error(f"{exc.__class__.__name__}: {exc.args[0]}")
+
+
+def _format_error(message: str) -> str:
+    parser = argparse.ArgumentParser()
+    settings.Settings.register_argparse_arguments(parser)
+    args, _ = parser.parse_known_args(sys.argv[1:])
+    color = not settings.Settings.from_argparse(args).no_color
```

Comment:
I don't think *any* argument parsing should be done in the function. Ideally, the configuration would be managed elsewhere, so that `color` is just an optional keyword argument, e.g.:

```suggestion
def _format_error(message: str, *, color: bool = True) -> str:
```